calculator-app/script.js

```diff
@@ -0,0 +1,127 @@
+const arr = [
+  ["C", "/", "*", "<"],
+  [7, 8, 9, "-"],
+  [4, 5, 6, "+"],
+  [1, 2, 3, "."],
+  ["(", 0, ")", "="],
+];
+
+// create container
+var app = document.createElement("div");
+app.className = "container";
+
+// create div for in / out
+var res = document.createElement("div");
+res.className = "inout";
+
+// create keys for in
+var btns = document.createElement("div");
+btns.className = "buttons";
+
+document.body.appendChild(app);
+var cont = document.querySelector(".container");
+cont.appendChild(res);
+cont.appendChild(btns);
+
+for (let i = 0; i < arr.length; i++) {
+  var row = document.createElement("div");
+  row.id = "r" + (i + 1);
+  arr[i].forEach((sym) => {
+    var key = document.createElement("div");
+    key.className = "key";
+    key.innerText = sym;
+    row.appendChild(key);
+  });
+  document.querySelector(".buttons").appendChild(row);
+}
+
+// retrieve all functional keys
+var r1 = document.getElementById("r1");
+var keys = r1.querySelectorAll(".key");
+var clrKey = keys[0];
+var delKey = keys[3];
+var eqKey = document.getElementById("r5").lastChild;
+clrKey.style.backgroundColor = "#e0314c";
+delKey.style.backgroundColor = "grey";
+eqKey.style.backgroundColor = "lightgrey";
+
+clrKey.id = "cls"; // assign ids for functional keys
+delKey.id = "del";
+eqKey.id = "eq";
+
+var inp = document.querySelector(".inout");
+
+btns = document.querySelectorAll(".key");
+
+var ops = ["+", "-", "*", "/"];
+var prec = ["(", ")"];
+
```

Comment:
Redundancy. These were already declared in ```arr```

Reply:
Accessing elements with indices will eat up more lines and this is readable indeed. Have to refactor large portion of the code if I consider this. Closing this issue 😉

Reply:
`var ops =[arr[i][j],...];
var prec = [arr[i][j]...]`

only way I see to do this 🤔 coz the elements in arr are along the columns and not rows 🤷‍♂️